### ISSUE TITLE
Yolo12-cls checkpoints not found in latest version

### ISSUE THREAD

**Issue by subh-775** [ultralytics/ultralytics#22887]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Seems like `v12-cls` isn't available in `ultralytics==8.3.7`

I ran into the issue while using the latest yolov12-classification model for image classification task.

![Image](https://github.com/user-attachments/assets/e196b19a-b8cc-4610-b5f0-5193b22eaeb9)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @subh-775, thank you for your interest in Ultralytics and for trying out the latest classification models 🚀  
This is an automated response 🤖 — an Ultralytics engineer will review your issue and follow up with more specific guidance soon.

From your description, it looks like you’re working with YOLO v12 classification checkpoints on `ultralytics==8.3.7` and encountering a “checkpoint not found” type of problem when using the `yolov12-classification` model.

To help the team investigate efficiently, please share a bit more detail:
- The exact command or Python snippet you are running
- The full error message or traceback (as text, not a screenshot, if possible)
- Which specific checkpoint name you are trying to use (for example, the full model string you pass to `YOLO(...)` or in the CLI)

In the meantime, we recommend a visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered ✅

If this is a 🐛 Bug Report related to missing YOLO v12 classification checkpoints or loading failures, it would be very helpful if you could also provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) that demonstrates the problem end‑to‑end.

If this is a custom training ❓ Question about using YOLO v12 for image classification (for example, with your own dataset), please include:
- A brief description of your dataset
- A couple of example images (or description of classes)
- Any relevant training logs
And verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) 💡

Join the Ultralytics community where it suits you best:
- For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧  
- Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/)  
- Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community

## Upgrade

Before debugging further, please make sure you are on the latest `ultralytics` package (and that all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) are installed) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/), and then re-test your issue:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
YOLO12 only has pretrained detection models

For other tasks, there are no pretrained models. You can still train them but you need to use `yolo12n-cls.yaml` not `.pt`

**Comment by subh-775** [comment]
thanks men `.yaml` is working for `mode=train`

**Comment by glenn-jocher** [comment]
Nice, glad that works — just note there are no official YOLO12-cls pretrained `.pt` weights, so for plug-and-play classification we recommend the YOLO11 `*-cls` models (see the [Image Classification docs](https://docs.ultralytics.com/tasks/classify/)), and for YOLO12-cls you’ll use your own trained checkpoint for inference.
